static/app/components/modals/metricWidgetViewerModal.tsx

```diff
@@ -0,0 +1,319 @@
+        </Header>
+        <Body>
+          <Query
+            widget={metricWidgetQueryParams}
+            projects={selection.projects}
+            onChange={handleChange}
+            contextMenu={
+              <ContextMenu
+                metricsQuery={{
+                  mri: metricWidgetQueryParams.mri,
+                  query: metricWidgetQueryParams.query,
+                  op: metricWidgetQueryParams.op,
+                  groupBy: metricWidgetQueryParams.groupBy,
+                  projects: selection.projects,
+                  datetime: selection.datetime,
+                  environments: selection.environments,
+                }}
+              />
+            }
+          />
+          <MetricWidget
+            widget={metricWidgetQueryParams}
+            onChange={(_, data) => {
+              handleChange(data);
+            }}
+            datetime={selection.datetime}
+            projects={selection.projects}
+            environments={selection.environments}
+          />
+          <MetricDetails widget={metricWidgetQueryParams} />
+        </Body>
+        <Footer>
+          <ButtonBar gap={1}>
+            <Button
+              onClick={() => {
+                closeModal();
+                router.push(
+                  getDdmUrl(organization.slug, {
+                    widgets: [metricWidgetQueryParams],
+                    project: selection.projects,
+                    ...selection.datetime,
+                    environment: selection.environments,
+                  })
+                );
+              }}
+            >
+              {t('Open in Metrics')}
+            </Button>
+            <Button
+              onClick={() => {
+                closeModal();
+              }}
+            >
+              {t('Close')}
+            </Button>
+            <Button
+              color="primary"
+              priority="primary"
+              onClick={() => {
+                handleSubmit();
+                closeModal();
+              }}
+            >
+              {t('Save changes')}
+            </Button>
+          </ButtonBar>
+        </Footer>
+      </OrganizationContext.Provider>
+    </Fragment>
+  );
+}
+export function ContextMenu({
+  metricsQuery,
+}: {
+  metricsQuery: MetricsQuery;
+}) {
+  const organization = useOrganization();
+  const router = useRouter();
+
+  const createAlert = useMemo(
+    () => getCreateAlert(organization, metricsQuery),
+    [metricsQuery, organization]
+  );
+
+  const items = useMemo<MenuItemProps[]>(() => {
+    const customMetric = !isCustomMetric({mri: metricsQuery.mri});
+    const addAlertItem = {
+      leadingItems: [<IconSiren key="icon" />],
+      key: 'add-alert',
+      label: t('Create Alert'),
+      disabled: !createAlert,
+      onAction: () => {
+        trackAnalytics('ddm.create-alert', {
+          organization,
+          source: 'widget',
+        });
+        Sentry.metrics.increment('ddm.dashboards.widget.alert');
```

Comment:
imho we should not add the ddm analytics events to dashboard.
I guess, we have the same problem when it comes to the query builder 🤔